ibis/common/rich_reprs.py

```diff
@@ -0,0 +1,333 @@
+import datetime
+from functools import singledispatch
+from math import isfinite
+
+import pandas as pd
+import rich
+from rich.align import Align
+from rich.console import Console
+from rich.text import Text
+
+import ibis
+import ibis.expr.datatypes as dt
+
+console = Console()
+# A console with all color/markup disabled, used for `__repr__`
+simple_console = Console(force_terminal=False)
+
+
+@singledispatch
```

Comment:
I like this approach!

Could you please move this file to something like `expr/types/pretty.py`? `ibis.common` should not be aware of the types defined under `ibis.expr`.

Reply:
Done.